### PR TITLE
Fixes graphs stuttering

### LINKED ISSUE #251: graphs stuttering
it should have a smooth motion

https://youtu.be/YrMwZUHzpBA

### PR DESCRIPTION
**Pull request by stsdc**
Should fix #251 

Call for tests. Has to be run  for a couple of hours, because previous implementation crushed a lot after some time.
@davidak @camellan  @ryonakano

### CODE REVIEW

**Comment by ryonakano** [comment]
@stsdc Hi, I'd like to test this branch but `ninja` command failed with the following error messages:

```bash
ryonakano@ryonakano-pc:~/Projects/monitor$ meson build --prefix=/usr
The Meson build system
Version: 0.53.2
Source dir: /home/ryonakano/Projects/monitor
Build dir: /home/ryonakano/Projects/monitor/build
Build type: native build
Project name: com.github.stsdc.monitor
Project version: 0.11.0
C compiler for the host machine: cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0")
C linker for the host machine: cc ld.bfd 2.34
Vala compiler for the host machine: valac (valac 0.48.19)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program sassc found: YES (/usr/bin/sassc)
Submodule 'subprojects/live-chart' (https://github.com/lcallarec/live-chart.git) registered for path 'live-chart'
Cloning into '/home/ryonakano/Projects/monitor/subprojects/live-chart'...
Submodule path 'live-chart': checked out '5f8ddce715f6a9130955bc5f08437050265a317c'

|Executing subproject live-chart method meson 
|
|Project name: live-chart
|Project version: 1.9.1
|C compiler for the host machine: cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0")
|C linker for the host machine: cc ld.bfd 2.34
|Vala compiler for the host machine: valac (valac 0.48.19)
|Library m found: YES
|Found pkg-config: /usr/bin/pkg-config (0.29.1)
|Run-time dependency gtk+-3.0 found: YES 3.24.20
|Run-time dependency gee-0.8 found: YES 0.20.3
|Build targets in project: 9
|Subproject live-chart finished.

Run-time dependency glib-2.0 found: YES 2.64.6
Run-time dependency granite found: YES 6.1.2
Dependency gtk+-3.0 found: YES 3.24.20 (cached)
Dependency gee-0.8 found: YES 0.20.3 (cached)
Run-time dependency gio-2.0 found: YES 2.64.6
Run-time dependency gobject-2.0 found: YES 2.64.6
Run-time dependency libgtop-2.0 found: YES 2.40.0
Run-time dependency libwnck-3.0 found: YES 3.36.0
Run-time dependency wingpanel found: YES 3.0.0
Run-time dependency gdk-x11-3.0 found: YES 3.24.20
Run-time dependency udisks2 found: YES 2.8.4
Run-time dependency libhandy-1 found: YES 1.2.0
Configuring config.h using configuration
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Library m found: YES
Library posix found: YES
Program meson/post_install.py found: YES (/home/ryonakano/Projects/monitor/meson/post_install.py)
Build targets in project: 26

com.github.stsdc.monitor 0.11.0

  Subprojects
    live-chart: YES

Found ninja-1.10.0 at /usr/bin/ninja
ryonakano@ryonakano-pc:~/Projects/monitor$ cd build/
ryonakano@ryonakano-pc:~/Projects/monitor/build$ ninja 
[1/201] Generating theme-monitor-light with a custom command.
FAILED: monitor-light.css 
/usr/bin/sassc .././data/stylesheet/monitor-light.scss monitor-light.css
Error: File to import not found or unreadable: ../../subprojects/stylesheet/src/palette.
        on line 1:1 of ../data/stylesheet/_index.scss
        from line 3:1 of ../data/stylesheet/monitor-light.scss
>> @import "../../subprojects/stylesheet/src/palette";

   ^
[2/201] Generating theme-monitor-dark with a custom command.
FAILED: monitor-dark.css 
/usr/bin/sassc .././data/stylesheet/monitor-dark.scss monitor-dark.css
Error: File to import not found or unreadable: ../../subprojects/stylesheet/src/palette.
        on line 1:1 of ../data/stylesheet/_index.scss
        from line 3:1 of ../data/stylesheet/monitor-dark.scss
>> @import "../../subprojects/stylesheet/src/palette";

   ^
[3/201] Generating theme-monitor-retrofit with a custom command.
FAILED: monitor-retrofit.css 
/usr/bin/sassc .././data/stylesheet/monitor-retrofit.scss monitor-retrofit.css
Error: File to import not found or unreadable: ../../subprojects/stylesheet/src/palette.
        on line 1:1 of ../data/stylesheet/monitor-retrofit.scss
>> @import "../../subprojects/stylesheet/src/palette";

   ^
[10/201] Compiling Vala source ../subprojects/live-chart/src/area.vala ../sub...tic/static_grid.vala ../subprojects/live-chart/src/static/static_points.vala.
ninja: build stopped: subcommand failed.
```

This also happens in the latest `master` and `dev` for me, so it would not due to the changes in this branch. And maybe I'm missing something because the CI passess. How can I fix these errors and build correctly?

**Comment by stsdc** [comment]
@ryonakano please use
```
git clone --recursive git://github.com/foo/bar.git
```

File: src/Resources/GPU.vala [hunk 1]
```diff
@@ -17,29 +17,13 @@ public class Monitor.GPU : Object {
         }
     }
 
-    public int percentage {
-        get {
-            return int.parse (get_sysfs_value ("/sys/class/drm/card0/device/gpu_busy_percent"));
-        }
-    }
+    public int percentage;
```
Comment (ryonakano) [inline src/Resources/GPU.vala]:
```suggestion
    public int percentage { get; private set; }
```

This property has had only a getter which means read-only. I guess we should keep that to be potentially substituted out of this class.

File: src/Resources/GPU.vala [hunk 2]
```diff
@@ -17,29 +17,13 @@ public class Monitor.GPU : Object {
...
 
-    public int memory_percentage {
-        get {
-            return int.parse (get_sysfs_value ("/sys/class/drm/card0/device/mem_busy_percent"));
-        }
-    }
+    public int memory_percentage;
```
Comment (ryonakano) [inline src/Resources/GPU.vala]:
```suggestion
    public int memory_percentage { get; private set; }
```

Same here.

File: src/Resources/GPU.vala [hunk 3]
```diff
@@ -17,29 +17,13 @@ public class Monitor.GPU : Object {
...
 
-    public int memory_vram_used {
-        get {
-            return int.parse (get_sysfs_value ("/sys/class/drm/card0/device/mem_info_vram_used"));
-        }
-    }
+    public int memory_vram_used;
```
Comment (ryonakano) [inline src/Resources/GPU.vala]:
```suggestion
    public int memory_vram_used { get; private set; }
```

File: src/Resources/GPU.vala [hunk 4]
```diff
@@ -17,29 +17,13 @@ public class Monitor.GPU : Object {
...
 
-    public double temperature {
-        get {
-            return double.parse (temperatures.get ("edge").input) / 1000;
-        }
-    }
+    public double temperature;
```
Comment (ryonakano) [inline src/Resources/GPU.vala]:
```suggestion
    public double temperature { get; private set; }
```

**Review by ryonakano — COMMENTED** [review]
I don't test for a couple of hours (I'll do that tomorrow afternoon), but meanwhile I left some code nitpickings :sweat_smile:

**Comment by stsdc** [comment]
@ryonakano all done. Thank You

**Comment by camellan** [comment]
https://user-images.githubusercontent.com/7396198/142731984-02d3deb9-3e77-4ff6-bf0b-5b43965e7374.mp4

The program runs for 2 hours. Everything is unchanged.

**Comment by camellan** [comment]
+ 2 hours. Everything is stable.

**Comment by ryonakano** [comment]
I did also test it for about 1 hour and nothing went wrong:

Screenshot from 2021-11-21 16-26-52

**Pull request merged**